lib/internal/Magento/Framework/DB/Ddl/Table.php

```diff
@@ -427,13 +428,23 @@ public function addForeignKey($fkName, $column, $refTable, $refColumn, $onDelete
             default:
                 $onDelete = self::ACTION_NO_ACTION;
         }
+        switch ($onUpdate) {
+            case self::ACTION_CASCADE:
+            case self::ACTION_RESTRICT:
+            case self::ACTION_SET_DEFAULT:
+            case self::ACTION_SET_NULL:
+                break;
+            default:
+                $onUpdate = self::ACTION_NO_ACTION;
+        }
 
         $this->_foreignKeys[$upperName] = [
             'FK_NAME' => $fkName,
             'COLUMN_NAME' => $column,
             'REF_TABLE_NAME' => $refTable,
             'REF_COLUMN_NAME' => $refColumn,
-            'ON_DELETE' => $onDelete
+            'ON_DELETE' => $onDelete,
+            'ON_UPDATE' => $onUpdate
```

Comment:
@xi-ao  @okorshenko ,
Regarding how Magento works right now, and how data structures stored for Magento entities (we generate "surrogate" Primary Key on which relation entities are dependent upon). These surrogate keys - are artificial, auto-incremented fields. DataBase generates and assigns them for entities by itself. So, there are no business cases when there is a valid business case when we need to change/update this Primary Key. It's just identifier for our entities.

The only difference we have in Magento is Product SKU. SKU - is a "natural" primary key for entity. But we don't have internal implementation for SKU as Primary Database field, and still use Product Id (surrogate key) for these purposes, and all other entities which related to Product have a reference by Product Id.
Referential integrity using Foreign Key constraint is much faster from the performance point of view, if integer type is used for this relationship linkage than string.

So, we could consider this not as a bug, but as a feature request. Because we have Framework component responsible for Database access. And because this component should provide general support of MySQL adapter, it should support "ON UPDATE".

But we can't consider this as a bug, because in Magento we neither use  nor planning to use this functionality.